plugins/unicodelib/unicodelib.cpp

```diff
@@ -719,6 +720,42 @@ unsigned unicodeEditDistanceV4(UnicodeString & left, UnicodeString & right, unsi
     return da[mask(leftLen-1)][rightLen-1];
 }
 
+void excludeLastWord(RuleBasedBreakIterator& bi, UnicodeString & source)
+{
+    bi.setText(source);
+    int32_t idx = bi.last();
+    int32_t wordidx = 0;
+    int32_t wordBeginning = 0;
+    int32_t wordEnd = bi.last();
+    while (idx != 0)
+    {
+        int breakType = bi.getRuleStatus();
+        if (breakType != UBRK_WORD_NONE)
+        {
+            // Exclude spaces, punctuation, and the like.
+            //   A status value UBRK_WORD_NONE indicates that the boundary does
+            //   not start a word or number.
+            ++wordidx;
+            wordBeginning = bi.previous();
+            if (bi.getRuleStatus() != UBRK_WORD_NONE)
+            {
+                ++wordidx;
+            }
+            if (bi.previous() == bi.first() && wordidx == 1)
```

Comment:
Actually documentation says:  
The character index of the preceding text boundary, or UBRK_DONE if all text boundaries have been returned.

I'm not sure what you are testing here.